Update deadlocks configuration setting in SQL Server guide (#32310)
DataDog/documentation: content/en/database_monitoring/guide/sql_deadlock.md

2. In the Datadog Agent, enable deadlocks in `sqlserver.d/conf.yaml`.
```yaml

>   collect_deadlocks: #prior to agent version 7.70, the setting was named deadlocks_collection

Reviewer: ```suggestion
  collect_deadlocks: # Renamed from deadlocks_collection in Agent version 7.70.
```
